wmfdata/hive.py

```diff
@@ -14,44 +14,26 @@ def run(cmds, fmt = "pandas", spark_master='yarn', app_name='wmfdata', spark_con
 
     if fmt not in ["pandas", "raw"]:
         raise ValueError("The `fmt` should be either `pandas` or `raw`.")
-
     if type(cmds) == str:
         cmds = [cmds]
 
-    # Check whether user has authenticated with Kerberos:
-    klist = subprocess.call("klist")
-    if klist == 1:
-        raise OSError(
-            "You do not have Kerberos credentials. " +
-            "Authenticate using `kinit` or run your script as a keytab-enabled user."
-        )
-    elif klist != 0:
-        raise OSError("There was an unknown issue checking your Kerberos credentials.")
-
-    result = None
-
-    spark = get_spark_session(spark_master, app_name, spark_config)
-
-    # TODO figure out how to handle multiple commands
+    spark_session = spark.get_session(spark_master, app_name, spark_config)
 
-    cmd = cmds[0]
-    resultDf = spark.sql(cmd)
+    result_DF = None
+    for cmd in cmds:
+        cmd_result = spark_session.sql(cmd)
+        
+        # If the result has columns, the command was a query and therefore results-producing.
+        # If not, it was a DDL or DML command and not results-producing.
+        if len(cmd_result.columns) > 0:
+            result_DF = cmd_result
     if fmt == 'pandas':
-        return resultDf.toPandas()
+        result = result_DF.toPandas()
```

Comment:
Won't this overwrite result for each cmd and only return the last result?

Reply:
Yes, that's the intention (as described by the function docstring 😊). I added the ability to run multiple commands in the same function call because, when we used impyla, that was the only way to set a session configuration setting before running a query.